controllers/capa_utils.go

```diff
@@ -41,3 +51,46 @@ func isRoleUsedElsewhere(ctx context.Context, ctrlClient client.Client, roleName
 
 	return false, err
 }
+
+func removeFinalizer(ctx context.Context, k8sClient client.Client, object client.Object, role string) error {
+	logger := log.FromContext(ctx)
```

Comment:
We should pass the logger from the caller, so that `WithValues(...)` info is retained. Or not log at all in this helper function, since the outermost caller is supposed to log any errors.

Reply:
I'd rather we just put the logger back when we modify it to avoid passing it around everywhere. I've done it for the controller but not the service. There we don't pass the context for some reason, so I want to avoid touching more of this code.

Reply:
I've also removed some of the values because they're already included if you use the logger from the context that controller-runtime passes to Reconcile

Reply:
Yes, that's implementation detail. As user, I basically want to see the values on each log line. If that's working right now with your changes, we're done with this here 😉. Sad that there are so many ways of doing this – we might never get it consistent 😆.